app/grandchallenge/components/views.py

```diff
@@ -288,3 +283,80 @@ def get_success_url(self):
 
     def get_success_message(self, cleaned_data):
         return f"{self.object._meta.verbose_name.title()} was successfully deleted"
+
+
+class BaseModelOptions(TextChoices):
+    ARCHIVE = Archive._meta.model_name
+    READER_STUDY = ReaderStudy._meta.model_name
+
+
+class CivSetListView(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, PaginatedTableListView

```

Comment:
Why are we using object permissions for a list view?

Reply:
We're checking if the user has change/use permission to the base object (archive or reader study) here. I took this over from how the list views used to be implemented since that made sense to me. Would it be better to use `PermissionListMixin` and check for permissions on each listed object?

Reply:
Yes, absolutely!

Reply:
It should always be this way btw, that way we control our rules about what users can do in one place (`assign_permissions`) and then they views do not have any logic about who can do what with what.